src/deepsparse/server/openai_server.py

```diff
@@ -175,6 +212,107 @@ async def abort_request() -> None:
             )
             return response
 
+        @app.post(
+            "/v1/completions",
+            tags=["model", "inference"],
+            response_model=CompletionResponse,
+        )
+        async def create_completion(raw_request: Request):
+            request = CompletionRequest(**await raw_request.json())
+            _LOGGER.debug(f"Received completion request: {request}")
+
+            model = request.model
+
+            pipeline = app.model_to_pipeline.get(model)
+            if not pipeline:
+                return create_error_response(
+                    HTTPStatus.BAD_REQUEST, f"{model} is not available"
+                )
+
+            request_id = f"cmpl-{random_uuid()}"
+            created_time = int(time.time())
+            prompt = request.prompt
+
+            try:
+                sampling_params = dict(
+                    num_return_sequences=request.n,
+                    presence_penalty=request.presence_penalty,
+                    frequency_penalty=request.frequency_penalty,
+                    temperature=request.temperature,
+                    top_p=request.top_p,
+                    top_k=request.top_k,
+                    stop=request.stop,
+                    max_tokens=request.max_tokens,
+                    stream=request.stream,
+                )
+            except ValueError as e:
+                return create_error_response(HTTPStatus.BAD_REQUEST, str(e))
+
+            result_generator = OpenAIServer.generate(
+                prompt=prompt,
+                request_id=request_id,
+                generation_kwargs=sampling_params,
+                pipeline=pipeline,
+            )
+
+            async def abort_request() -> None:
+                await pipeline.abort(request_id)
```

Comment:
Does the pipeline object have a abort method? Isn't it a property of DeepSparseOpenAIEngine